packages/angular/cli/models/parser.ts

```diff
@@ -109,19 +109,29 @@ function _getOptionFromName(name: string, options: Option[]): Option | undefined
 }
 
 
-function _assignOption(
+function _assignOption({
+  arg,
+  nextArg,
+  options,
+  parsedOptions,
+  leftovers,
+  ignored,
+  errors,
+}: {
   arg: string,
```

Comment:
change the first argument from being the actual argument to the index of the argument.  You can then access the current argument and next argument by using the existing `args` array.

Also, have it return the new index.  This would eliminate the extra conditionals present in the calling function.

Reply:
I don't think this is possible, as the value for "arg" is computed rather than from the array for one of the function calls (I'll leave a comment pointing out the call in question)